public/client/TracyProfiler.cpp

```diff
@@ -3063,6 +3097,62 @@ bool Profiler::CommitData()
     return ret;
 }
 
+char* Profiler::SafeCopyProlog( const char* data, size_t size )
+{
+    bool success = true;
+    char* buf = m_safeSendBuffer;
+#ifndef NDEBUG
+    assert( !m_inUse.exchange(true) );
+#endif
+
+    if( size > m_safeSendBufferSize ) buf = (char*)tracy_malloc( size );
```

Comment:
What's the typical histogram of buffer sizes? How often is a larger temporary buffer allocated?

Reply:
I did a test on a simple application and got the following:
  | 1 << ? | Count | Comment
-- | -- | -- | --
  | [0] | 0 | 
  | [1] | 0 | 
  | [2] | 8 | 
  | [3] | 85 | 
  | [4] | 280 | 
  | [5] | 685 | 
  | [6] | 863 | 
  | [7] | 917 | 
  | [8] | 856 | 
  | [9] | 799 | 
  | [10] | 619 | 
  | [11] | 449 | 
  | [12] | 239 | 
  | [13] | 124 | 
  | [14] | 58 | 
  | [15] | 34 | 
  | [16] | 17 | 
  | [17] | 2 | (exceeds buffer)

So at 64k in my simple test only 2 requests required temporary heap allocation as they exceeded the buffer size